openpype/hosts/maya/plugins/publish/validate_arnold_scene_source.py

```diff
@@ -35,17 +34,29 @@ def _get_nodes_data(self, nodes):
             if parent:
                 parents.append(parent)
 
-            nodes_by_name[node_split[-1]] = node
-            for shape in cmds.listRelatives(node, shapes=True):
-                nodes_by_name[shape.split("|")[-1]] = shape
+            node_name = node.rsplit("|", 1)[-1].rsplit(":", 1)[-1]
+
+            # Check for same same nodes, which can happen in different
+            # hierarchies.
+            if node_name in nodes_by_name:
+                same_named_nodes.append((node, nodes_by_name[node_name]))
```

Comment:
If the node name would e.g. appear five times then this would 4 times report each entry together with the first entry. Is that what is intended?

Might make more sense to log it as `Found nodes with the same name {name}: {nodes}`?